Add AI usage guidelines to CONTRIBUTING guide (#1179)
epiforecasts/EpiNow2: .github/CONTRIBUTING.md

Config/Needs fields, so use pak or remotes instead if you need the
additional developer tools.

## AI-assisted contributions

> We welcome contributions created with the assistance of AI tools (such as Claude, GitHub Copilot, ChatGPT, or similar). However, contributors using AI assistance must be transparent about it. For example, if AI tools were used to generate a substantial portion of your contribution (code, documentation, or tests), indicate this in your PR description and indicate where you need more attention.

Reviewer: what do you mean by

> indicate where you need more attention.

Config/Needs fields, so use pak or remotes instead if you need the
additional developer tools.

## AI-assisted contributions

We welcome contributions created with the assistance of AI tools (such as Claude, GitHub Copilot, ChatGPT, or similar). However, contributors using AI assistance must be transparent about it. For example, if AI tools were used to generate a substantial portion of your contribution (code, documentation, or tests), indicate this in your PR description and indicate where you need more attention.

> Currently, AI-tools automatically [co-author git commits](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) but if that's not the case for the tool you use, ensure to acknowledge AI contributions in a transparent way.

Author: ```suggestion

Currently, AI-tools automatically [co-author git commits](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) but if that's not the case for the tool you use, ensure to acknowledge AI contributions in a transparent way.
```

### Transparency and disclosure

*   **Disclose AI usage in PRs**: If AI tools were used to generate a substantial portion of your contribution (code, documentation, or tests), indicate this in your PR description. You can add a note at the end of your PR summary:

    ```markdown
    🤖 Generated with [Tool Name](https://example.com)
    ```

*   **Minor AI assistance doesn't require disclosure**: Using AI for autocompletion, spell-checking, or minor suggestions doesn't require explicit disclosure. Use your judgement—if AI generated most of the content, disclose it.

### Contributor responsibility

*   **You are responsible for all submitted code**: Regardless of how code was generated, you are fully responsible for understanding, testing, and maintaining your contribution.

*   **Review AI-generated code thoroughly**: AI can produce code that looks correct but contains subtle bugs, security issues, or doesn't follow our coding standards. Always:
    *   Read and understand every line of AI-generated code
    *   Test thoroughly using `devtools::test()` and `devtools::check()`

>     *   Check coding standards with `lintr::lint_package()`

Author: Yes, some of these are better run on the CI but I think it's good practice to run your (inexpensive) tests and linters locally before pushing?

    *   Verify documentation is accurate and follows our style

*   **Request reviews appropriately**: If your PR is substantially AI-generated and you haven't yet reviewed it thoroughly yourself, add a note at the top of your PR description:

    ```markdown
    **This PR contains substantial AI-generated content. I will complete my review
    before requesting maintainer review.**
    ```

    Then, once you've completed your own review and are confident in the changes, update your PR to indicate it's ready for team review.

### AI-assisted PR descriptions

> When writing PR descriptions for AI-assisted contributions, follow this template:

Author: Fixed in 3fac96ba58d8658bd2a882c3a9c58fb54cd6a92e

    The title of your PR should briefly describe the change.
    The body of your PR should contain `Fixes #issue-number`.

*   If your PR is a work in progress, [mark it as a draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) until it's ready, then update your PR to indicate it's ready for team review.

> *   For user-facing changes, add a bullet to the top of `NEWS.md` (i.e. just below the first header). Follow the style described in <https://style.tidyverse.org/news.html>.

Reviewer: ```suggestion
*   For user-facing changes, add a bullet to the top of `NEWS.md` in the section that corresponds to the type of change. Follow the style described in <https://style.tidyverse.org/news.html>.
```

#### What happens after submitting a PR?

*   PRs are reviewed by the team before they are merged. The review process only begins after the continuous integration checks, which have to be manually triggered by a maintainer for first-time contributors, have passed.

> *   You should expect to see reviews from our AI code reviewer, [CodeRabbit](https://www.coderabbit.ai/), which will point out some initial issues to fix. Be sure to address review comments that you think are accurate and reply with reasons why other comments won't be fixed. Provide commits to support your fixes.

Reviewer: ```suggestion
*   You should expect to see reviews from an AI code reviewer, which will point out any issues to fix. Be sure to address review comments that you think are accurate and reply with reasons why other comments won't be fixed. Provide commits to support your fixes.
```
